workbench/src/renderer/components/Changelog/index.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from 'react-bootstrap/Button';
+import Modal from 'react-bootstrap/Modal';
+import { MdClose } from 'react-icons/md';
+import { useTranslation } from 'react-i18next';
+
+import pkg from '../../../../package.json';
+import { ipcMainChannels } from '../../../main/ipcMainChannels';
+
+const { ipcRenderer } = window.Workbench.electron;
+
+export default function Changelog(props) {
+    const { t } = useTranslation();
+    const [htmlContent, setHtmlContent] = useState('');
+
+    // Load HTML from external file (which is generated by Python build process).
+    useEffect(() => {
+        async function loadHtml() {
+            const baseUrl = await ipcRenderer.invoke(ipcMainChannels.BASE_URL)
+            fetch(`${baseUrl}/changelog.html`)
+            .then(response => response.text())
+            .then(htmlString => {
+                // Find the section whose heading explicitly matches the current version.
+                const versionStr = pkg.version;
+                const escapedVersionStr = versionStr.split('.').join('\\.');
+                const sectionRegex = new RegExp(
+                    `<section.*?>[\\s]*?<h1>${escapedVersionStr}\\b[\\s\\S]*?</h1>[\\s\\S]*?</section>`
+                );
+                const sectionMatches = htmlString.match(sectionRegex);
+                if (sectionMatches && sectionMatches.length) {
+                    let latestVersionSection = sectionMatches[0];
+                    const linkRegex = /<a\shref/g;
+                    // Ensure all links open in a new window and are styled with a relevant icon.
+                    latestVersionSection = latestVersionSection.replaceAll(
+                        linkRegex,
+                        '<a target="_blank" class="link-external" href'
```

Comment:
Do you think it's possible to have these open in the default browser, instead of an electron window? That usually requires calling `electron.shell.openExternal` via a click event (we have `ipcMainChannels.OPEN_EXTERNAL_URL`). But given the static html, I'm not sure if that's possible. Curious if you have any ideas, but it's not critical to change.

Reply:
Good question. I did come across this pattern in our code, but I didn't follow it for two reasons:
1. I have a knee-jerk reaction whenever I see a click handler added to a link. Links are supposed to "just work" the way they are, and forcing them to behave in a non-default manner is often—though not always—a sign that there might be a simpler solution.
2. I misunderstood the purpose of `OPEN_EXTERNAL_URL` (even though it's clearly documented, so that's on me!) and thought it only opened links in a new (Electron) window. So when these links opened in a new window with nothing more than `target="_blank"`, I figured that was all I needed.

Looking around on the web just now, I came across some references to [`setWindowOpenHandler`](https://www.electronjs.org/docs/latest/api/web-contents#contentssetwindowopenhandlerhandler) in the Electron API. I'm not 100% sure it does what you're suggesting, or that it'd be "better" than our existing solution (heck, for all I know, it just adds click handlers to all the links behind the scenes), but I was curious to know if you had ever seen it or even tried it.

Reply:
> 1. I have a knee-jerk reaction whenever I see a click handler added to a link. Links are supposed to "just work" the way they are, and forcing them to behave in a non-default manner is often—though not always—a sign that there might be a simpler solution.

This is a good point!

>Looking around on the web just now, I came across some references to [setWindowOpenHandler](https://www.electronjs.org/docs/latest/api/web-contents#contentssetwindowopenhandlerhandler) in the Electron API. I'm not 100% sure it does what you're suggesting, or that it'd be "better" than our existing solution (heck, for all I know, it just adds click handlers to all the links behind the scenes), but I was curious to know if you had ever seen it or even tried it.

I never noticed this before. It looks potentially useful for standardizing some options we use when we want to open a new electron window. And I guess it can also "deny" the creation of the new electron window and do something else instead. The challenge might be that sometimes we want to open a new electron window, and sometimes we might prefer a default-browser window.

Cases where we use an electron window:
* About menu pages
* User's Guide link -- though this one could be changed to a non-electron window if we want to, it links to local html files, not the web version.

There are a couple reasons why I think it's a nicer experience to use the user's default browser for links on the web:
1. I might already be logged into github in my browser, so opening a github link there is more convenient than opening it in an electron window.
2. Our electron windows don't have any of the useful browser UI features, such as a Back button, address bar field, etc. So if I wanted to start on the linked page and then continue browsing around github from there, I'm better off in my own browser than in electron. 

I think these are the two main reasons why we setup that `OPEN_EXTERNAL_URL` handler and use it for the FAQ link to the Forum. But given the static html of the changelog, it would be more complicated to add a click handler.

Reply:
These are all great points. I'm going to see if I can make it work.

Reply:
Updated! All the links in the changelog modal now open in the browser.